sympy/solvers/bivariate.py

```diff
@@ -176,22 +179,85 @@ def _lambert(eq, x):
     p, den = den.as_coeff_Mul()
     e = exp(num/den)
     t = Dummy('t')
-    args = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
-
     # calculating solutions from args
-    for arg in args:
+    if p == 1:
+        t = e
+        args = [d/(a*b)*t]
+    elif domain.is_real:
+        ind_ls = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
+        args = []
+        j = -1
+        for i in ind_ls:
+            j += 1
+            if not isinstance(i,int):
+                if not i.has(I):
+                    args.append(ind_ls[j])
+            elif isinstance(i,int):
+                args.append(ind_ls[j])
+    else:
+        args = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
+    if len(args) == 0:
+        return S.EmptySet
+    # for simplicity defined two functions
+    # Lambert_Real_Braches , lambert_helper
+    def Lambert_Real_Braches():
         for k in lambert_real_branches:
-            w = LambertW(arg, k)
-            if k and not w.is_real:
+            rhs2 = -c/b + (a/d)*LambertW(arg, k)
+            if rhs2.has(I):
                 continue
-            rhs = -c/b + (a/d)*w
+            sol1 = lambert_helper(rhs2)
+        return sol1,rhs2
+
+    def lambert_helper(rhs):
+    # to avoid repetitive code
+        for xu in xusolns:
+            sol.append(xu.subs(u, rhs))
+        return sol
+
+    domainReal = domain.is_subset(S.Reals)
+    integer = Symbol('integer',integer=True)
+    for arg in args:
+        rhs1 = -c/b + (a/d)*LambertW(arg)
+        LambertReal = LambertW(arg).is_real
```

Comment:
A capitalised name like `LambertReal` would normally be used for a class. Since this is a boolean variable it would be better to call this `lambert_real`.

Reply:
Same goes for e.g. `domainReal`